static/app/components/smartSearchBar/index.tsx

```diff
@@ -914,6 +1102,37 @@ class SmartSearchBar extends React.Component<Props, State> {
     );
   }
 
+  /**
+   * Updates autocomplete dropdown items and autocomplete index state
+   *
+   * @param groups Groups that will be used to populate the autocomplete dropdown
+   */
+  updateAutoCompleteStateMultiHeader = (groups: AutocompleteGroup[]) => {
+    const {hasRecentSearches, maxSearchItems, maxQueryLength} = this.props;
+    const query = this.state.query;
+    const queryCharsLeft =
+      maxQueryLength && query ? maxQueryLength - query.length : undefined;
+    const searchGroups = {
+      searchGroups: [] as SearchGroup[],
+      flatSearchItems: [] as SearchItem[],
+      activeSearchItem: -1,
+    };
+    for (const group of groups) {
+      const {searchItems, recentSearchItems, tagName, type} = group;
+      const searchGroup = createSearchGroups(
+        searchItems,
+        hasRecentSearches ? recentSearchItems : undefined,
+        tagName,
+        type,
+        maxSearchItems,
+        queryCharsLeft
+      );
+      searchGroups.searchGroups.push(...searchGroup.searchGroups);
+      searchGroups.flatSearchItems.push(...searchGroup.flatSearchItems);
+    }
```

Comment:
If you wanna do this functionally you could do this

```tsx
const searchGroups = groups
  .map(({searchItems, recentSearchItems, tagName, type}) =>
    createSearchGroups(
      searchItems,
      hasRecentSearches ? recentSearchItems : undefined,
      tagName,
      type,
      maxSearchItems,
      queryCharsLeft
    )
  )
  .reduce(
    (acc, item) => ({
      searchGroups: [...acc.searchGroups, ...item.searchGroups],
      flatSearchItems: [...acc.flatSearchItems, ...item.flatSearchItems],
    }),
    {
	  searchGroups: [] as SearchGroup[],
      flatSearchItems: [] as SearchItem[],
      activeSearchItem: -1,
	}
  );
```

maybe harder to read idk

Reply:
seems like we lean towards doing stuff functionally so I'll change to map/reduce